### PR TITLE
added firehose

### PR DESCRIPTION
**Pull request by ruchiramoitra**
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-transformer/191)

### CODE REVIEW

File: __tests__/data/firehose.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (prabrishac) [inline __tests__/data/firehose.json:1]: If you want to commit this file please provide data too.

Comment (ruchiramoitra) [reply __tests__/data/firehose.json]: done

File: v0/destinations/firehose/transform.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+function getDeliveryStreamMapTo(event) {
+  const config = event.destination.Config;
```
Comment (arnab-p) [inline v0/destinations/firehose/transform.js:2]: Do we need this extra line? Could be just

`const { mapEvents } = event.destination.Config;`

File: v0/destinations/firehose/transform.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const { message } = event;
+
+  const { mapEvents } = config;
+  let check = true;
+  let deliveryStreamMapFrom;
+  let deliveryStreamMapTo = null;
+  mapEvents.forEach(mapEvent => {
```
Comment (arnab-p) [inline v0/destinations/firehose/transform.js:9]: Could it be simplified as

```
const fromName = message.event || message.type || "*";

// This block can go to the Util
const hashMap = {};
mapEvents.forEach(ev => {
hashMap[ev.from] = ev.to;
});
// This block ends here

const toName = hashMap[fromName];

// check whether toName is undefined

```

File: v0/destinations/firehose/transform.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  if (getDeliveryStreamMapTo(event) == null) {
+    throw new Error(
+      "No delivery stream set for event",
+      event.message.event || event.message.type
+    );
+  } else {
+    const result = {
+      message: event.message,
+      userId: event.message.userId || event.message.anonymousId,
+      deliveryStreamMapTo: getDeliveryStreamMapTo(event)
```
Comment (arnab-p) [inline v0/destinations/firehose/transform.js:47]: Should we persist the result of `getDeliveryStreamMapTo`? Duplicate method call.

File: v0/destinations/firehose/transform.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    };
+    return result;
```
Comment (arnab-p) [inline v0/destinations/firehose/transform.js:49]: Go for inline return?

File: v0/destinations/firehose/transform.js (hunk 2 above)
Comment (ruchiramoitra) [reply v0/destinations/firehose/transform.js]: done

File: v0/destinations/firehose/transform.js (hunk 3 above)
Comment (ruchiramoitra) [reply v0/destinations/firehose/transform.js]: done

File: v0/destinations/firehose/transform.js (hunk 4 above)
Comment (ruchiramoitra) [reply v0/destinations/firehose/transform.js]: done

File: v0/destinations/firehose/transform.js (hunk 5 above)
Comment (ruchiramoitra) [reply v0/destinations/firehose/transform.js]: done

**Pull request merged**